packages/engine.test/src/testView.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Engine } from "@c11/engine.react";
+import { generateImage } from 'jsdom-screenshot';
+
+jest.useFakeTimers();
+
+export async function testView(name: string, { state, View, props }) {
+    test(name, async () => {
+        const root = document.createElement("div")
+        const title = document.createElement("h1")
+        title.innerHTML = name
+        
+        document.body.innerHTML = ""
+        document.body.appendChild(title)
+        document.body.appendChild(root)
+      
+        const engine = new Engine({
+          state: {
+              initial: state
+          },
+          view: {
+            element: <View
+              {...props}
+            />,
+            root
+          },
+        });
+      
+        jest.runAllTimers();
+      
+        const screenshot = await generateImage();
```

Comment:
Should allow the developer to pass a configuration object. This is especially useful when a component might end up triggering network requests which can then be subbed.
https://www.npmjs.com/package/jsdom-screenshot#optionsintercept

Reply:
as per the engine pattern, every request would be mocked so there is no reason to intercept anything

Reply:
I don't think this is about the mocks you can give in the views input. From what I read this will run in a headless chrome instance using puppeteer. This means that even resources will try to load. For example:
```
const foo: view = () => <img src="http://someplace.com/image.jpg" />
```
This will make a request and the net effect will be that it slows down (either until the resource is loaded or it times out).
In the eventuality that the tests will be run for regression, this will yield different results based on network conditions, or the volatility of the resource in question.

This is just one example, there can be many other cases.